### DISCUSSION TITLE
Is it possible to call the resolution resampler from the CLI?

### DISCUSSION THREAD

**Discussion by ToshY** [arch1t3cht/Aegisub#144] [Q&A] [answered]
I'm looking for a way to either use, extract or rewrite the code that is used to resample subtitles. While I've find the files that are responsible for resampling to be [`resolution_resampler.cpp`](https://github.com/arch1t3cht/Aegisub/blob/feature/src/) and [`resolution_resampler.h`](https://github.com/arch1t3cht/Aegisub/blob/feature/src/resolution_resampler.h), I have no idea if I can use that functionality as standalone functionality.

Is there any way to either call that functionality from the command line, or extract the functionality to make it completely standalone, or even rewrite it to another language (e.g. Python)?

> [!NOTE]
> I have zero knowledge regarding anything related to C(++).

**Comment by arch1t3cht** [comment, marked as answer]
There is a separate project called [aegisub-cli](https://github.com/Myaamori/aegisub-cli) that also allows resampling from the command line. It's effectively abandoned (the maintainer having been inactive for years), but it works.

I have some in-progress work on integrating aegisub-cli's features into Aegisub itself (see the `cli` branch), but I was distracted with more important upstream work and since then didn't have as much time to work on it anymore. Maybe one day I'll be able to finish it.

**Reply by ToshY** [reply]
Thanks for the quick response!

> There is a separate project called [aegisub-cli](https://github.com/Myaamori/aegisub-cli) that also allows resampling from the command line. It's effectively abandoned (the maintainer having been inactive for years), but it works.

Good to know that atleast something like that already exists. I'll definately check it out (probably try to dockerize it if possible).

> I have some in-progress work on integrating aegisub-cli's features into Aegisub itself (see the cli branch), but I was distracted with more important upstream work and since then didn't have as much time to work on it anymore. Maybe one day I'll be able to finish it.

It's okay, take all the time you need. Your work here is much appreciated 🥇
